homeassistant/components/sensor/simulated.py

```diff
@@ -0,0 +1,140 @@
+"""
+Adds a simulated sensor.
+
+For more details about this platform, refer to the documentation at
+https://home-assistant.io/components/sensor.simulated/
+"""
+import datetime as datetime
+import math
+import random
+import logging
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = datetime.timedelta(seconds=1)
+ICON = 'mdi:chart-line'
+
+CONF_UNIT = 'unit'
+CONF_AMP = 'amplitude'
+CONF_MEAN = 'mean'
+CONF_PERIOD = 'period'
+CONF_PHASE = 'phase'
+CONF_FWHM = 'spread'
+CONF_SEED = 'seed'
+
+DEFAULT_NAME = 'simulated'
+DEFAULT_UNIT = 'value'
+DEFAULT_AMP = 1
+DEFAULT_MEAN = 0
+DEFAULT_PERIOD = 60
+DEFAULT_PHASE = 0
+DEFAULT_FWHM = 0
+DEFAULT_SEED = 999
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_UNIT, default=DEFAULT_UNIT): cv.string,
+    vol.Optional(CONF_AMP, default=DEFAULT_AMP): vol.Coerce(float),
+    vol.Optional(CONF_MEAN, default=DEFAULT_MEAN): vol.Coerce(float),
+    vol.Optional(CONF_PERIOD, default=DEFAULT_PERIOD): cv.positive_int,
+    vol.Optional(CONF_PHASE, default=DEFAULT_PHASE): vol.Coerce(float),
+    vol.Optional(CONF_FWHM, default=DEFAULT_FWHM): vol.Coerce(float),
+    vol.Optional(CONF_SEED, default=DEFAULT_SEED): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the simulated sensor."""
+    name = config.get(CONF_NAME)
+    unit = config.get(CONF_UNIT)
+    amp = config.get(CONF_AMP)
+    mean = config.get(CONF_MEAN)
+    period = config.get(CONF_PERIOD)
+    phase = config.get(CONF_PHASE)
+    fwhm = config.get(CONF_FWHM)
+    seed = config.get(CONF_SEED)
+    random.seed(seed)
```

Comment:
What if seed is None?

Reply:
Config Val doesn't allow None, but should I still catch?